core/services/ocr2/plugins/promwrapper/plugin.go

```diff
@@ -0,0 +1,141 @@
+package promwrapper
+
+import (
+	"context"
+	"math/big"
+	"time"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promauto"
+	"github.com/smartcontractkit/libocr/offchainreporting2/types"
+)
+
+var _ types.ReportingPlugin = &promPlugin{}
+
+var (
+	promQuery = promauto.NewHistogramVec(
+		prometheus.HistogramOpts{
+			Name:    "ocr2_reporting_plugin_query_duration",
+			Help:    "The amount of time elapsed during the OCR2 plugin's Query() method",
+			Buckets: []float64{}, // TODO: figure out buckets
+		},
+		[]string{"evmChainID", "plugin"},
+	)
+	promObservation = promauto.NewHistogramVec(
+		prometheus.HistogramOpts{
+			Name:    "ocr2_reporting_plugin_observation_duration",
+			Help:    "The amount of time elapsed during the OCR2 plugin's Observation() method",
+			Buckets: []float64{}, // TODO: figure out buckets
+		},
+		[]string{"evmChainID", "pluginName"},
+	)
+	promReport = promauto.NewHistogramVec(
+		prometheus.HistogramOpts{
+			Name:    "ocr2_reporting_plugin_report_duration",
+			Help:    "The amount of time elapsed during the OCR2 plugin's Report() method",
+			Buckets: []float64{}, // TODO: figure out buckets
+		},
+		[]string{"evmChainID", "pluginName"},
+	)
+	promShouldAcceptFinalizedReport = promauto.NewHistogramVec(
+		prometheus.HistogramOpts{
+			Name:    "ocr2_reporting_plugin_should_accept_finalized_report_duration",
+			Help:    "The amount of time elapsed during the OCR2 plugin's ShouldAcceptFinalizedReport() method",
+			Buckets: []float64{}, // TODO: figure out buckets
+		},
+		[]string{"evmChainID", "pluginName"},
+	)
+	promShouldTransmitAcceptedReport = promauto.NewHistogramVec(
+		prometheus.HistogramOpts{
+			Name:    "ocr2_reporting_plugin_should_transmit_accepted_report_duration",
+			Help:    "The amount of time elapsed during the OCR2 plugin's ShouldTransmitAcceptedReport() method",
+			Buckets: []float64{}, // TODO: figure out buckets
+		},
+		[]string{"evmChainID", "pluginName"},
+	)
+	promClose = promauto.NewHistogramVec(
+		prometheus.HistogramOpts{
+			Name:    "ocr2_reporting_plugin_close_duration",
+			Help:    "The amount of time elapsed during the OCR2 plugin's Close() method",
+			Buckets: []float64{}, // TODO: figure out buckets
+		},
+		[]string{"chainType", "chainID", "pluginName"},
```

Comment:
I think you should use these labels (`[]string{"chainType", "chainID", "pluginName"}`) for all the above metrics as well.

Reply:
:heavy_plus_sign:  No preference for `pluginName` vs `plugin`, but should align them all too

Reply:
Hmm yeah `plugin` might be better, I think it's clear that (in this context) that it's referring to a name.